sqlglot/optimizer/merge_subqueries.py

```diff
@@ -92,7 +99,7 @@ def merge_ctes(expression, leave_tables_isolated=False):
     return expression
 
 
-def merge_derived_tables(expression, leave_tables_isolated=False):
+def merge_derived_tables(expression: exp.Expression, leave_tables_isolated=False):
```

Comment:
Here too, let's just do another good pass on these, I'm assuming the same may have happened elsewhere.